llvm/lib/Transforms/Yk/ControlPoint.cpp

```diff
@@ -173,7 +173,7 @@ class YkControlPoint : public ModulePass {
 
     // Create the new control point.
```

Comment:
Can we add a comment giving an idea of the type signature of the control point we're creating?

Reply:
Sure. I'm not too sold on the current design (though it does keep us going). The control point currently returns a bool to tell the interpreter whether to continue, or return, which is a bit crude. We also need to return the actual return value via the control point which we are currently not doing. Two options I can think of: a) return a struct, e.g.  `{retval, continue}`, or pass one of them into the control point as a pointer, which the stopgap interpreter then writes to.

Reply:
I'm fine with this as an interim design, while we work out if it's general enough or not. I just would like us to document what the control point will "look like" (so to speak) so our future selves don't have to reverse engineer all the LLVM function calls :)

Reply:
Fixed in [eda3b57](https://github.com/ykjit/ykllvm/pull/22/commits/eda3b57f4570e970def8c4138f4a9e73988ceb20)